addons/crm/models/crm_lead.py

```diff
@@ -430,16 +430,21 @@ def write(self, vals):
 
         return write_result
 
-    def _write_probability(self, vals):
+    def _write_probability(self, vals, force_compute=False):
         compute = False
-        fields_to_check = ['tag_ids', 'stage_id', 'team_id'] + self._pls_get_safe_fields()
-        for field, value in vals.items():
-            if field in fields_to_check:
-                compute = True
-                break
+
+        if force_compute:
+            compute = True
+        else:
+            fields_to_check = ['tag_ids', 'stage_id', 'team_id'] + self._pls_get_safe_fields()
+            for field, value in vals.items():
+                if field in fields_to_check:
+                    compute = True
+                    break
 
         if not compute:
             return
+
```

Comment:
Correct but I think you can compact a little more :

```
if not force_compute:
    fields_to_check = ['tag_ids', 'stage_id', 'team_id'] + self._pls_get_safe_fields()
    for field, value in vals.items():
        if field in fields_to_check:
            force_compute = True
            break
if not force_compute:
    return
# and the rest..
```
and you can remove the last return at the end